iceoryx_utils/source/posix_wrapper/mutex.cpp

```diff
@@ -50,6 +50,16 @@ mutex::mutex(bool f_isRecursive)
                                       &attr,
                                       PTHREAD_PRIO_NONE)
                           .hasErrors();
+    if (robust == Robust::ON)
+    {
+        isInitialized &= !cxx::makeSmartC(pthread_mutexattr_setrobust,
```

Comment:
I think this call is not supported for Mac OS and therefore has to be reimplemented and I am not aware if Windows is supporting robust mutexes?! 

But maybe you can follow up this link: https://stackoverflow.com/questions/24946886/is-there-pthread-mutex-robust-equivalent-in-mac-os-x or contact me and we can work together on a solution.